games-arcade/cdogs-sdl/cdogs-sdl-0.10.1.ebuild

```diff
@@ -0,0 +1,44 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake xdg
+
+DESCRIPTION="Classic overhead run-and-gun game"
+HOMEPAGE="https://cxong.github.io/cdogs-sdl/"
+SRC_URI="https://github.com/cxong/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2+ BSD-2 CC0-1.0 CC-BY-3.0 CC-BY-SA-3.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="
+	dev-python/protobuf-python
+	media-libs/libsdl2[haptic,opengl]
+	media-libs/sdl2-image[png]
+	media-libs/sdl2-mixer[vorbis,wav]
+	net-libs/enet:1.3="
+RDEPEND="${DEPEND}"
+
+src_prepare() {
+	rm -f doc/{COPYING.,LICENSE.,license.,README_DATA.md}* || die
+
+	# Disable cdogs-sdl-editor configure
+	sed -i -e '/add_subdirectory(cdogsed)/d' src/CMakeLists.txt || die
+	# Disable cdogs-sdl-editor build
+	sed -i -e 's/add_executable(cdogs-sdl-editor/add_executable(cdogs-sdl-editor EXCLUDE_FROM_ALL/' src/CMakeLists.txt || die
```

Comment:
I'd prefer a patch for this but if you must use `sed` then you can be smarter about it.

```bash
sed -i -e 's/add_executable(cdogs-sdl-editor/\0 EXCLUDE_FROM_ALL/' src/CMakeLists.txt || die
```

Same below.

Reply:
Maybe better to use `cmake_comment_add_subdirectory` from cmake eclass (at least for `sed -i -e '/add_subdirectory(cdogsed)/d' src/CMakeLists.txt || die` call)?

Another solution:

According to src/Cmakelists.txt, cdogs-sdl-editor controlled by ${GCW0} variable (used for crosscompile). Logic is if this is not cross-compile environment, ${GCW0} will set to 0 and editor is disabled. So, you can do little tricky hax:

```
# Not tested, need to check quotes and escaping
sed -i -e 's/if(NOT "${GCW0}")/if(FALSE)/g' src/CMakeLists.txt || die
```
Now you'll have sections like
```
if(FALSE)
# Some cdogs-sdl-editor stuff
endif()
```

Reply:
I went ahead and submitted a PR upstream for a patch to make the editor build conditional so that we don't need to worry about these hacky `sed` lines: https://github.com/cxong/cdogs-sdl/pull/693

Reply:
Excellent, thanks!